.pipelines/templates/e2e-test-kind-arc.yaml

```diff
@@ -0,0 +1,75 @@
+jobs:
+  - job: e2e_arc_kind
+    variables:
+    - name: AZURE_ENVIRONMENT_FILEPATH
+      value: /etc/kubernetes/custom_environment.json
+    - name: VOLUME_NAME
+      value: cloudenvfile-vol
+    - group: csi-secrets-store-e2e-kind
+    steps:
+    - script: |
+        az login -i > /dev/null
+        az account set -s=$(SUBSCRIPTION_ID)
+      displayName: "az login"
+    - script: |
+        az extension add --name connectedk8s
+        az extension add --name k8s-extension
+      displayName: "add cli extensions"
+    - script: |
+        make install-helm install-kubectl setup-kind
+      displayName: "setup KinD"
+    - script: |
+        clusterName=sscd-arc-e2e-$(openssl rand -hex 6)
+        echo "##vso[task.setvariable variable=AZURE_CLUSTER_NAME]$clusterName"
+        echo "cluster name is set to - $clusterName"
+      displayName: "set cluster name"
+      condition: succeeded()
+    - script: |
+        az group create -n ${AZURE_CLUSTER_NAME} -l $(AZURE_CANARY_LOCATION)
+
+        az connectedk8s connect -n ${AZURE_CLUSTER_NAME} -g ${AZURE_CLUSTER_NAME} --no-wait
+
+        echo "verifying cluster connectivity..."
+        for i in $(seq 1 25); do
```

Comment:
If the cluster is never connected, this step will not error. We should error here if the connectivityStatus doesn't become `Connected` after retries